cmd/session-clear.go

```diff
@@ -0,0 +1,126 @@
+package cmd
+
+import (
+	"encoding/json"
+
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/console"
+	"github.com/minio/minio/pkg/probe"
+)
+
+var sessionClear = cli.Command{
+	Name:            "clear",
+	Usage:           "Clear a previously saved session.",
+	Action:          mainSessionClear,
+	Before:          setGlobalsFromContext,
+	Flags:           globalFlags,
+	HideHelpCommand: true,
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+  
+USAGE:
+  {{.HelpName}} [SESSION-ID/all]
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. Clear session.
+	 $ {{.HelpName}} clear ygVIpSJs
+  2. Clear all sessions.
+	 $ {{.HelpName}} clear all
+`,
+}
+
+// clearSessionMessage container for clearing session messages.
+type clearSessionMessage struct {
+	Status    string `json:"success"`
+	SessionID string `json:"sessionId"`
+}
+
+// String colorized clear session message.
+func (c clearSessionMessage) String() string {
+	msg := "Session `" + c.SessionID + "`"
+	var colorizedMsg string
+	switch c.Status {
+	case "success":
+		colorizedMsg = console.Colorize("ClearSession", msg+" cleared successfully.")
+	case "forced":
+		colorizedMsg = console.Colorize("ClearSession", msg+" cleared forcefully.")
+	}
+	return colorizedMsg
+}
+
+// JSON jsonified clear session message.
+func (c clearSessionMessage) JSON() string {
+	clearSessionJSONBytes, e := json.Marshal(c)
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+
+	return string(clearSessionJSONBytes)
+}
+
+// clearSession clear sessions.
+func clearSession(sid string) {
+	if sid == "all" {
+		for _, sid := range getSessionIDs() {
+			session, err := loadSessionV8(sid)
+			fatalIf(err.Trace(sid), "Unable to load session `"+sid+"`.")
+
+			fatalIf(session.Delete().Trace(sid), "Unable to remove session `"+sid+"`.")
+
+			printMsg(clearSessionMessage{Status: "success", SessionID: sid})
+		}
+		return
+	}
+
+	if !isSessionExists(sid) {
+		fatalIf(errDummy().Trace(sid), "Session `"+sid+"` not found.")
+	}
+
+	session, err := loadSessionV8(sid)
+	if err != nil {
+		// `mc session clear <broken-session-id>` assumes that user is aware that the session is unuseful
+		// and wants the associated session files to be removed
+		removeSessionFile(sid)
+		removeSessionDataFile(sid)
+		printMsg(clearSessionMessage{Status: "forced", SessionID: sid})
+		return
+	}
+
```

Comment:
We are okay to crash here if session is nil . Defensive coding in this manner can mask bugs avoid it.